From: Nico Pache <npache@redhat.com>
Subject: kunit/bug.c: fix backtrace-supression-test failure as a module
Date: Tue, 7 Jul 2026 12:54:05 -0600

When running the backtrace-supression-test as a module most cases produce
failures.

Upon further investigating the issue presents itself due to the ifdef that
wraps the kunit_is_suppressed_warning, which is not compiled in if using
CONFIG_KUNIT_MODULE.

Fix this by switching to IS_ENABLED(CONFIG_KUNIT) which properly checks
for either CONFIG_KUNIT_MODULE or CONFIG_KUNIT.

Link: https://lore.kernel.org/20260707185405.571277-1-npache@redhat.com
Fixes: bbc960d009a6 ("kunit: Add backtrace suppression self-tests")
Signed-off-by: Nico Pache <npache@redhat.com>
Cc: Alessandro Carminati <acarmina@redhat.com>
Cc: Dan Carpenter <error27@gmail.com>
Cc: David Gow <david@davidgow.net>
Cc: Guenetr Roeck <linux@roeck-us.net>
Cc: Kees Cook <kees@kernel.org>
Cc: Nico Pache <npache@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/bug.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/bug.c~kunit-bugc-fix-backtrace-supression-test-failure-as-a-module
+++ a/lib/bug.c
@@ -219,7 +219,7 @@ static enum bug_trap_type __report_bug(s
 	no_cut   = bug->flags & BUGFLAG_NO_CUT_HERE;
 	has_args = bug->flags & BUGFLAG_ARGS;
 
-#ifdef CONFIG_KUNIT
+#if IS_ENABLED(CONFIG_KUNIT)
 	/*
 	 * Before the once logic so suppressed warnings do not consume
 	 * the single-fire budget of WARN_ON_ONCE().
_
